cpp/src/arrow/csv/writer_test.cc

```diff
@@ -189,6 +198,12 @@ std::vector<WriterTestParams> GenerateTestCases() {
             /*expected_output*/ "", expected_status_no_quotes_with_structural(error_val)};
   };
 
+  // Schema/expected message for delimiter test
+  auto schema_custom_delimiter = schema({field("a", int64()), field("b", int64())});
+  auto batch_custom_delimiter = R"([{"a": 42, "b": -12}])";
+  auto expected_output_delimiter_tabs = "42\t-12\n";
+  auto expected_output_delimiter_pipe = "42|-12\n";
```

Comment:
Looks duplicated, one test should be enough.

Reply:
Removed duplicate case.